### PR TITLE
modify feedback widget behaviour

### PR DESCRIPTION
**Pull request by marcellamalune**
## Description

- Add `isExpanded` state to FeedbackWidget component
- Set a 30sec timeout as a side effect to change `isExpanded` state to true and clear timeout on unmount
- Use media query to apply style changes only on desktop devices
- Apply 0.5 transition timing for width and border-radius
- Reset `isExpanded` to false when `isOpen` is reset too
- Import `Text` component from ChakraUI
- Truncate text while expanding to show a smooth animation

## Related Issue

https://github.com/ethereum/ethereum-org-website/issues/8298

### CODE REVIEW

File: src/components/FeedbackWidget.tsx [hunk 1]
```diff
@@ -220,8 +240,23 @@ const FeedbackWidget: React.FC<IProps> = ({ className }) => {
...
     <>
-      <FixedDot onClick={handleOpen} bottomOffset={bottomOffset} id="dot">
+      <FixedDot
+        onClick={handleOpen}
+        bottomOffset={bottomOffset}
+        isExpanded={isExpanded}
+        id="dot"
+      >
         <StyledFeedbackGlyph />
+        <Text
```
Comment (pettinarip) [inline src/components/FeedbackWidget.tsx:250]: I think we could wrap this `Text` with a transition like `ScaleFade` (https://chakra-ui.com/docs/components/transitions/usage#scalefade-transition) and delay it a bit.

IMO the animation should be:
1. Expand the button to the left (keeping the icon on the left)
2. Once the expand animation finish, we display the Text with a fade in or scale fade animation

**Review by pettinarip — CHANGES_REQUESTED** [review]
Thanks for the PR @marcellamalune. I think the basic animation is working but there are some problems when the expanding animation starts. Take a look at this to see it clearly:
https://user-images.githubusercontent.com/468158/198043249-d6e3e1f4-5c69-4c76-bb2f-c46ed4ea9608.mp4

**Comment by pettinarip** [comment]
This is looking nice so far @marcellamalune. Wanted to check with @konopkja what he thinks about it? should the animation be faster? or is it ok as it is?

**Comment by konopkja** [comment]
> 

looks nice! agree, the animation could be twice as fast. other than that it looks good.

**Comment by marcellamalune** [comment]
Thank you for your feedback @pettinarip @konopkja! I speeded up the animation.

**Comment by konopkja** [comment]
> * Deploy preview

ok this is better. @pettinarip im good with this.

**Comment by corwintines** [comment]
Screen Shot 2022-10-31 at 12 01 38 PM

Hey @marcellamalune!

Noticing this isn't working on mobile.

**Comment by marcellamalune** [comment]
> Screen Shot 2022-10-31 at 12 01 38 PM
> 
> Hey @marcellamalune!
> 
> Noticing this isn't working on mobile.

Sorry about that, I screwed up something with the latest changes without realizing. Now it's fixed.

**Comment by corwintines** [comment]
Thanks for fixing @marcellamalune. Will get to review this week, but I did add the `hacktoberfest-accepted` label to get you credit for this still.

File: src/components/FeedbackWidget.tsx [hunk 2]
```diff
@@ -33,6 +36,17 @@ const FixedDot = styled(NakedButton)<{
...
   }
+  @media (min-width: ${({ theme }) => theme.breakpoints.l}) {
+    width: ${({ isExpanded }) => (isExpanded ? "15rem" : "3rem")};
+    border-radius: ${({ isExpanded }) => (isExpanded ? "50px" : "50%")};
+    #feedback-wrapper {
+      width: ${({ isExpanded }) => (isExpanded ? "13.5rem" : "3rem")};
+      position: ${({ isExpanded }) => (isExpanded ? "absolute" : "flex")};
+    }
+    #expanded-prompt {
+      display: ${({ isExpanded }) => (isExpanded ? "flex" : "none")};
```
Comment (pettinarip) [inline src/components/FeedbackWidget.tsx:47]: I don't love the idea of using ids to achieve this. Why can't we implement this logic using style props?

Can't we do this like
```tsx
<Text ... width={{ base: "3rem", lg: isExpanded ? "13.5rem" : "3rem" }}>
  <Box
    as={Translation}
    id="feedback-card-prompt-page"
    display={{ base: "flex", lg: isExpanded ? "flex" : "none" }}
  />
</Text>
```

**Review by pettinarip — CHANGES_REQUESTED** [review]
The overall result looks great right now @marcellamalune. My only concern is more about the implementation using ids, wondering if we could adjust a bit that.

**Comment by konopkja** [comment]
@pettinarip @marcellamalune can we push this forward?

File: src/components/FeedbackWidget.tsx (hunk 2 above)
Comment (marcellamalune) [reply src/components/FeedbackWidget.tsx]: @pettinarip thanks for the feedback and for the improvements you've made 🙂

**Review by pettinarip — APPROVED** [review]

**Review by corwintines — APPROVED** [review]
Thanks @marcellamalune and @pettinarip!

**Pull request merged**